### ISSUE TITLE
[camera] Audio disabled, but android.permission.RECORD_AUDIO is still included

### ISSUE THREAD

**Issue by octocat-1** [flutter/flutter#98326]
```dart
CameraController(
  enableAudio: false
);
```

The Flutter Camera package adds `android.permission.RECORD_AUDIO` permission to "AndroidManifest.xml", even when audio has been disabled for the camera controller.

I also don't understand the benefit of Flutter Camera automatically adding these permissions to "AndroidManifest.xml", instead of letting the developer edit the permissions.

**Comment by maheshmnj** [comment]
Hi @octocat-1, Thanks for filing the issue. I do see that the camera plugin uses the Audio permissions  in its [Androidmanifest](https://github.com/flutter/plugins/blob/2efdee1e7368cebdfccf88bbab6887156d2f4f7d/packages/camera/camera/android/src/main/AndroidManifest.xml)


> The Flutter Camera package adds android.permission.RECORD_AUDIO permission to "AndroidManifest.xml", even when audio has been disabled for the camera controller.

I am quite not sure if we can modify the permissions in runtime.

> I also don't understand the benefit of Flutter Camera automatically adding these permissions to "AndroidManifest.xml", instead of letting the developer edit the permissions.

I believe it is required for the camera plugin to function

Leaving this issue open for further insights from the team.

cc: @bparrishMines

**Comment by octocat-1** [comment]
Another reason the Flutter Camera package should not add the Camera and Audio permissions automatically to the AndroidManifest files, is because it leads the developer to think that because those permissions are enabled by the Flutter Camera package they do not need to explicitly define the permission for other packages like [Permission Handler](https://pub.dev/packages/permission_handler) that depend on that permission.

In other words, it leads the developer to create a dependency between the Flutter Camera package and other packages, which if the Flutter Camera package is uninstalled, would crash other packages that depend on those permissions.

**Comment by log02** [comment]
Any update on this ? Is there a way to remove the AUDIO permission ?

**Comment by Milou6** [comment]
I just modified my `CameraController` to disable audio with `enableAudio: false` as indicated [here](https://github.com/flutter/flutter/issues/31477), but Chrome is still asking for both Audio & Camera permissions.

**Comment by octocat-1** [comment]
As a temporary fix for Android I'm using:

```xml
    <uses-permission android:name="android.permission.RECORD_AUDIO"
        tools:node="remove" />
```

**Comment by YAHYA6K** [comment]
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    package="" xmlns:tools="http://schemas.android.com/tools">

    <uses-permission android:name="android.permission.RECORD_AUDIO"
        tools:node="remove" />

</mainfest>

**Comment by Arcttyx** [comment]
Any update on this?
Hack proposed does not work, at least in my case.
Users feel this permission as dangerous and dont want to use the camera.

**Comment by fourman-alex** [comment]
> ```dart
> CameraController(
>   // ...
>   enableAudio: false
> );
> ```
> 
> The Flutter Camera package adds `android.permission.RECORD_AUDIO` permission to "AndroidManifest.xml", even when audio has been disabled for the camera controller.
> 
> I also don't understand the benefit of Flutter Camera automatically adding these permissions to "AndroidManifest.xml", instead of letting the developer edit the permissions.

I agree that there should be a way to control the audio permission for use cases where audio is not required and thus the permission should not be requested. Having said that, `enableAudio` only controls whether audio should be recorded along side the video.
```
  /// Controls audio presence in recorded video.
  final bool enableAudio;
```

We do need a way to configure the plugin to not use audio when not required.

**Comment by camsim99** [comment]
https://developer.android.com/reference/android/content/Context#revokeSelfPermissionOnKill(java.lang.String) might be a relevant API we can use
